client/reader/follow-helpers.tsx

```diff
@@ -0,0 +1,42 @@
+/**
+ * External Dependencies
+ */
+import { escapeRegExp } from 'lodash';
+
+/**
+ * Internal Dependencies
+ */
+import {
+	getSiteName,
+	getSiteUrl,
+	getSiteDescription,
+	getSiteAuthorName,
+} from 'calypso/reader/get-helpers';
+
+type Follow = Record< string, unknown >;
+
+export function filterFollowsByQuery(
+	query: undefined | string,
+	follows: Array< Follow >
+): Array< Follow > {
+	if ( ! query ) {
+		return follows;
+	}
+
+	const phraseRe = new RegExp( escapeRegExp( query ), 'i' );
+
+	return follows.filter( ( follow ) => {
+		const feed = follow.feed;
+		const site = follow.site;
+		const siteName = getSiteName( { feed, site } );
+		const siteUrl = getSiteUrl( { feed, site } );
+		const siteDescription = getSiteDescription( { feed, site } );
+		const siteAuthor = getSiteAuthorName( site );
+
+		return (
+			`${ follow.URL }${ siteName }${ siteUrl }${ siteDescription }${ siteAuthor }`.search(
```

Comment:
might want to add separators between the parts here to prevent run-ons from matching. like a description of `crossword` and an author of `press` would match `wordpress`. just a space should be enough, but a | or a non-breaking-space would be even safer.